src/rewriter.coffee

```diff
@@ -661,6 +662,18 @@ exports.Rewriter = class Rewriter
       @detectEnd i + 1, condition, action
       return 1
 
+  # Convert TERMINATOR followed by && or || into a single LEADING_&& or
+  # LEADING_|| token to disambiguate grammar.
```

Comment:
Why not `LEADING_AND` and `LEADING_OR`? We prefer the English word names as a general rule in the codebase. Unless you need separate tokens for `LEADING_AND` and `LEADING_&&`?

Speaking of which, this should work for `and` as well as `&&`.

Reply:
@GeoffreyBooth sure, I used `LEADING_&&` to mimic the existing `&&` token type, but agreed it reads badly, updated

It works equivalently for `and` and `&&`, as by the time the rewriter/grammar sees them they're the same token type (and both variants are included in `INDENT_SUPPRESSOR`). A couple of the tests use `&&`/`||`